cpp/src/arrow/python/util/datetime.h

```diff
@@ -18,22 +18,178 @@
 #ifndef PYARROW_UTIL_DATETIME_H
 #define PYARROW_UTIL_DATETIME_H
 
+#include <algorithm>
+#include <sstream>
+
 #include <datetime.h>
+#include "arrow/status.h"
+#include "arrow/util/logging.h"
 #include "arrow/python/platform.h"
 
 namespace arrow {
 namespace py {
 
+// The following code is adapted from
+// https://github.com/numpy/numpy/blob/master/numpy/core/src/multiarray/datetime.c
```

Comment:
Could we add the NumPy license to our LICENSE.txt at the bottom? Maybe also remove the occurrence of NumPy from the NOTICE.txt. I am not 100% sure when we have to use NOTICE and when only append to LICENSE.txt. From looking at Apache Kudu it seems most of what is in NOTICE should be in the LICENSE.txt https://github.com/apache/kudu

Reply:
Ok, I added it to LICENSE.txt and removed it from NOTICE.txt.